### PR TITLE
feat: implement dependency resource_identifier

### PR DESCRIPTION
**Pull request by underfin**
## Summary

It used as an identifier to merge equal requests, it can reduce resolve same request for dependency.

## Test Plan

It should be tested for exits test.

### CODE REVIEW

**Review by hyf0 — DISMISSED** [review]
LGTM. @h-a-n-a cc

File: crates/node_binding/binding.d.ts [hunk 1]
```diff
@@ -976,11 +969,3 @@ export interface RawTrustedTypes {
   policyName?: string
 }
 
-export interface ThreadsafeNodeFS {
-  writeFile: (...args: any[]) => any
-  removeFile: (...args: any[]) => any
-  mkdir: (...args: any[]) => any
-  mkdirp: (...args: any[]) => any
-  removeDirAll: (...args: any[]) => any
-}
```
Comment (Boshen) [inline crates/node_binding/binding.d.ts:985]: Why is this removed again.

File: packages/rspack/tests/__snapshots__/StatsTestCases.test.ts.snap [hunk 2]
```diff
@@ -580,7 +554,6 @@ exports[`StatsTestCases should print correct stats for ignore-plugin 1`] = `
 exports[`StatsTestCases should print correct stats for ignore-plugin 2`] = `
 "./index.js [10] {main}
 ./locals/en.js [208] {main}
-<PROJECT_ROOT>/tests/statsCases/ignore-plugin/locals/./zh (missing) [665] {main}
```
Comment (Boshen) [inline packages/rspack/tests/__snapshots__/StatsTestCases.test.ts.snap:583]: Was this a bug before? Why is this removed?

File: crates/node_binding/binding.d.ts (hunk 1 above)
Comment (hyf0) [reply crates/node_binding/binding.d.ts:985]: napi-rs currently could not generate a stable `.d.ts` file.

Comment (underfin) [reply crates/node_binding/binding.d.ts:985]: It's my bad.

File: packages/rspack/tests/__snapshots__/StatsTestCases.test.ts.snap (hunk 2 above)
Comment (underfin) [reply packages/rspack/tests/__snapshots__/StatsTestCases.test.ts.snap:583]: This is a same request for next line `<PROJECT_ROOT>/tests/statsCases/ignore-plugin/locals/./zh.js`

**Review by h-a-n-a — APPROVED** [review]

**Review by hyf0 — APPROVED** [review]
